dotnet/src/SemanticKernel/CoreSkills/TextMemorySkill.cs

```diff
@@ -37,49 +39,90 @@ public class TextMemorySkill
     /// </summary>
     public const string KeyParam = "key";
 
+    /// <summary>
+    /// Name of the context variable used to specify the number of memories to recall
+    /// </summary>
+    public const string LimitParam = "limit";
+
     private const string DefaultCollection = "generic";
     private const string DefaultRelevance = "0.75";
+    private const string DefaultLimit = "1";
+
+    /// <summary>
+    /// Recall a specific memory
+    /// </summary>
+    /// <example>
+    /// SKContext[TextMemorySkill.KeyParam] = "countryInfo1"
+    /// {{memory.recall_specific }}
+    /// </example>
+    /// <param name="context">Contains the 'collection' containing the memory to recall and the `key` associated with it.</param>
+    [SKFunction("Recall a specific memory")]
+    [SKFunctionName("Recall_Specific")]
+    [SKFunctionContextParameter(Name = CollectionParam, Description = "Memories collection associated with the memory to recall",
+        DefaultValue = DefaultCollection)]
+    [SKFunctionContextParameter(Name = KeyParam, Description = "The key associated with the memory to recall")]
+    public async Task<string> RecallSpecificAsync(SKContext context)
+    {
+        var collection = context.Variables.ContainsKey(CollectionParam) ? context[CollectionParam] : DefaultCollection;
+        Verify.NotEmpty(collection, "Memory collection not defined");
+
+        var key = context.Variables.ContainsKey(KeyParam) ? context[KeyParam] : string.Empty;
+        Verify.NotEmpty(key, "Memory key not defined");
+
+        context.Log.LogTrace("Recalling memory from collection '{0}'", collection);
+
+        var memory = await context.Memory.GetAsync(collection, key);
+        return memory != null ? memory.Text : string.Empty;
+    }
 
     /// <summary>
-    /// Recall a fact from the long term memory
+    /// Recall up to N memories related to the input
     /// </summary>
     /// <example>
     /// SKContext["input"] = "what is the capital of France?"
     /// {{memory.recall $input }} => "Paris"
     /// </example>
-    /// <param name="ask">The information to retrieve</param>
-    /// <param name="context">Contains the 'collection' to search for information and 'relevance' score</param>
-    [SKFunction("Recall a fact from the long term memory")]
+    /// <param name="text">The input text to compare memories to</param>
+    /// <param name="context">Contains the 'collection' to search for the topic and 'relevance' score</param>
+    [SKFunction("Recall up to N memories related to the input")]
     [SKFunctionName("Recall")]
-    [SKFunctionInput(Description = "The information to retrieve")]
-    [SKFunctionContextParameter(Name = CollectionParam, Description = "Memories collection where to search for information", DefaultValue = DefaultCollection)]
+    [SKFunctionInput(Description = "The input text to compare memories to")]
```

Comment:
nit: "The text to search for related memories." or something similar?

Reply:
"The input text to find related memories for"?